cpu/msp430fxyz/periph/spi.c

```diff
@@ -213,18 +214,38 @@ int spi_release(spi_t dev)
     return 0;
 }
 
-int spi_transfer_byte(spi_t dev, char out, char *in)
+int spi_transfer_bytes(spi_t dev, char *out, char *in, unsigned int length)
 {
     (void)dev;
-    char tmp;
-    while (!(SPI_IF & SPI_IE_TX_BIT)) {}
-    SPI_DEV->TXBUF = (uint8_t)out;
-    while (!(SPI_IF & SPI_IE_RX_BIT)) {}
-    tmp = (char)SPI_DEV->RXBUF;
-    if (in) {
-        *in = tmp;
+
+    assert(out || in);
+
+    /* if we only send out data, we do this the fast way... */
+    if (!in) {
+        for (unsigned i = 0; i < length; i++) {
+            while (!(SPI_IF & SPI_IE_TX_BIT)) {}
+            SPI_DEV->TXBUF = (uint8_t)out[i];
+        }
+        while (!(SPI_IF & SPI_IE_RX_BIT)) {}
+        SPI_DEV->RXBUF;
```

Comment:
We need the following here, even though it's not the prettiest solution. I've read the manual several times and tested different configurations on (and between) z1 and telosb motes so I'm pretty convinced this is the right thing to do:
```
#ifndef SPI_USE_USCI
        while (!(SPI_IF & SPI_IE_TX_BIT) || !(SPI_IF & SPI_IE_RX_BIT)) {}
        SPI_DEV->RXBUF;
#else
        while (SPI_DEV->STAT & 1) {}
        SPI_DEV->RXBUF;
#endif
```
IIRC you already introduced a macro for checking bit no 1 (`BUSY` bit) in the status register for  devices **with** ` SPI_USE_USCI` but this was lost I guess.